api/include/opentelemetry/trace/global_provider.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <mutex>
```

Comment:
I don't think we can depend on `std::mutex` if we can't depend on e.g. `std::unique_ptr` :(

Reply:
Good to know! Is there a good resource to read about the ABI differences between different implementations of stdlib?

Reply:
Not that I'm aware of. :(

So far we've been careful not to use any types. e.g. no std::string, vector, unique_ptr, string_view (also it's not in C++11).

IMO atomic is fine because it doesn't affect memory layout.